ledger-service/db-backend/src/main/scala/com/digitalasset/http/dbbackend/Queries.scala

```diff
@@ -40,13 +40,29 @@ sealed abstract class Queries {
   /** for use when generating predicates */
   protected[this] val contractColumnName: Fragment = sql"payload"
 
+  /** Table names with prefix * */
+
+  val contractTableNameRaw = s"${tablePrefix}contract"
```

Comment:
I think it would be great to test that this is set everywhere it needs to be.
I think there are two options for this:
1. Some form of unit test over `initDatabaseDdls` that checks that everything is correctly prefixed.
2. An integration test that spins up two instances of the JSON API with different prefixes in the same database and checks that both work.
I’d probably go for 2 but your choice.

Reply:
Just wrote a test locally which does `2` and caught a bug, thanks for the heads up :smile:

Reply:
@cocreature would you be fine with me pushing the last fixes I made to the code and adding the test in a seperate pr? It turned out to be a bit more code :smile:

Reply:
sounds reasonable